game/theater/iadsnetwork/iadsnetwork.py

```diff
@@ -118,34 +119,48 @@ def skynet_nodes(self, game: Game) -> list[SkynetNode]:
             if game.iads_considerate_culling(node.group.ground_object):
                 # Skip culled ground objects
                 continue
-            try:
-                skynet_node = SkynetNode.from_group(node.group)
-                for connection in node.connections.values():
-                    if connection.ground_object.is_friendly(
-                        skynet_node.player
-                    ) and not game.iads_considerate_culling(connection.ground_object):
-                        skynet_node.connections[connection.iads_role.value].append(
-                            SkynetNode.dcs_name_for_group(connection)
-                        )
-                skynet_nodes.append(skynet_node)
-            except IadsNetworkException:
-                # Node not skynet compatible
-                continue
+
+            skynet_node = SkynetNode.from_group(node.group)
+            for connection in node.connections.values():
+                is_friendly = connection.ground_object.is_friendly(skynet_node.player)
+                is_culled = game.iads_considerate_culling(connection.ground_object)
+                if is_friendly and not is_culled:
+                    key = connection.iads_role.value
+                    name = SkynetNode.dcs_name_for_group(connection)
+                    skynet_node.connections[key].append(name)
+            skynet_nodes.append(skynet_node)
+
         return skynet_nodes
 
-    def update_tgo(self, tgo: TheaterGroundObject) -> None:
+    def _update_iads_building(
+        self, tgo: TheaterGroundObject, events: GameUpdateEvents
+    ) -> None:
+        if not isinstance(tgo, IadsBuildingGroundObject):
+            return
+
+        for node in self.nodes:
+            for cID, group in node.connections.items():
+                if group.ground_object is tgo:
+                    events.delete_iads_connection(cID)
```

Comment:
This seems upside-down... is there not a way to iterate connections directly on the tgo?

Reply:
Comm towers and power stations have no connections, they're not nodes. Nodes connect to them instead...